tools/clang/include/clang/SPIRV/SpirvModule.h

```diff
@@ -22,6 +22,23 @@ namespace spirv {
 class SpirvFunction;
 class SpirvVisitor;
 
+struct RichDebugInfo {
+  RichDebugInfo(SpirvDebugSource *src, SpirvDebugCompilationUnit *cu)
+      : source(src), compilationUnit(cu) {
+    scopeStack.push_back(cu);
+  }
+  RichDebugInfo() : source(nullptr), compilationUnit(nullptr), scopeStack() {}
+
+  // The HLL source code
+  SpirvDebugSource *source;
+
+  // The compilation unit (topmost debug info node)
+  SpirvDebugCompilationUnit *compilationUnit;
+
+  // Stack of lexical scopes
+  std::vector<SpirvDebugInstruction *> scopeStack;
```

Comment:
Shouldn't the lexical scopes be inside the compilation unit?  Or can there be lexical scopes that do not belong to a CU?

Reply:
All lexical scopes must be inside a CU except the CU itself. Since there can be multiple CUs, a different CU will not be inside a CU.